web_external/ImagesGallery/ImagesFilters.js

```diff
@@ -0,0 +1,219 @@
+        } else {
+            return undefined;
+        }
+    },
+
+    _rangeFacetFilterExpression: function (facetId, facetFilters) {
+        var filterExpression = _.chain(facetFilters)
+            .pick(function (binIncluded, binLabel) {
+                return binIncluded === false;
+            })
+            .map(function (binIncluded, binLabel) {
+                // Parse range labels, to yield an array of numeric range arrays
+                var rangeMatches = binLabel.match(/\[([\d.]+) - ([\d.]+)\)/);
+                var lowBound = parseFloat(rangeMatches[1]);
+                var highBound = parseFloat(rangeMatches[2]);
+                return [lowBound, highBound];
+            })
+            .reduce(function (allRanges, curRange) {
+                // Combine adjacent ranges
+                var prevRange = _.last(allRanges);
+                // Compare the previous high bound and the current low bound, checking for
+                // the special case of the first element where there's no "prevRange"
+                if (prevRange && prevRange[1] === curRange[0]) {
+                    // If they match, combine the current high bound with the previous low
+                    // bound
+                    var combinedRange = [prevRange[0], curRange[1]];
+                    // Pop the previous range, and add the combined range
+                    return allRanges.slice(0, -1).concat([combinedRange]);
+                } else {
+                    // If there's no match, just push the current range onto the list
+                    return allRanges.concat([curRange]);
+                }
+            }, [])
+            .map(function (range) {
+                // Convert each range into a string expression
+                var lowBoundExpression = 'not (' +
+                    isic.SerializeFilterHelpers._stringToHex(facetId) +
+                    ' >= ' +
+                    range[0] +
+                    ')';
+                var highBoundExpression = 'not (' +
+                    isic.SerializeFilterHelpers._stringToHex(facetId) +
+                    ' < ' +
+                    range[1] +
+                    ')';
+                return '(' + lowBoundExpression + ' or ' + highBoundExpression + ')';
+            })
+            .value()
+            // Combine all expressions
+            .join(' and ');
+        return '(' + filterExpression + ')';
+    },
+
+    _categoricalFacetFilterExpression: function (facetId, facetFilters) {
+        var excludedBinLabels = _.chain(facetFilters)
+            .pick(function (binIncluded, binLabel) {
+                return binIncluded === false;
+            })
+            .keys()
+            .map(isic.SerializeFilterHelpers._stringToHex)
+            .value();
+        // TODO: Could use "facetId + ' in ' + includedBinLabels"" if most are excluded
+        return '(' +
+            isic.SerializeFilterHelpers._stringToHex(facetId) +
```

Comment:
Putting quote characters around the identifier should allow us to simplify this code (won't need `_stringToHex()` anymore hopefully).